### DISCUSSION TITLE
Remove the recently written chars

### DISCUSSION THREAD

**Discussion by cnicodeme** [ueberdosis/tiptap#3438] [General Questions] [answered]
Hi!

I've made a custom Mention plugin, that allows me to run a command that will insert a specific text when executed.

I'm using the Mention plugin along with the Suggestion and I have set the "char" to be "!". This allows me to write something like:

```!bye```

And it will replace it with 

```Thank you for reaching us.

Best regards,
John```

The issue I'm facing is that when I execute `editor.chain().focus().insertContent('Thank you for reaching....').run()`, what I initially wrote (`!bye`) is still present.

How can I remove it? I haven't found a way to get the current selection.

From the place where the "insertContent" code is used, I can have access to the editor, what was written (the query)).

Thank you for your help.

**Comment by cnicodeme** [comment, marked as answer]
Welp ... I just found the solution and it was way easier than what I went on to try ...

The `props` sent by the Suggestion plugin includes the `range` property, so all I had to do was:

```
editor
                    .chain()
                    .focus()
                    .setTextSelection(this.range)
                    .deleteSelection()
                    .run()
```

**Reply by AbhijitSP11** [reply]
I want to edit the character after entering a suggestion.  What I am currently using is - 

           editor .chain().focus().insertContentAt(range, [
              {
                type: this.name,
                attrs: props,
              },
              {
                type: "text",
                text: " ",
              },
            ])
            .run();``

**Reply by pghanem** [reply]
@abhi-j @cnicodeme could you guys elaborate on how you did this or share more code? I'm new and wondering where you put these command chains to achieve this.

**Reply by cnicodeme** [reply]
The commands we share are run from the object you get when you create a Tiptap editor, so depending on where you store your instance, you can run it from anywhere.

```
const editor = new Editor({
  element: document.querySelector('.element'),
  extensions: [
    Document,
    Paragraph,
    Text,
  ],
  content: '<p>Example Text</p>',
  autofocus: true,
  editable: true,
  injectCSS: false,
})

editor
    .chain()
    .focus()
    .setTextSelection(this.range)
    .deleteSelection()
    .run()
```
